posydon/utils/common_functions.py

```diff
@@ -1203,13 +1239,15 @@ def He_MS_lifetime(mass):
         He MS time duration in yr.
 
     """
-    if mass < 2.0:
+    if mass <=0.0:
+        raise ValueError(f"Too low mass: {mass}")
+    elif mass < 2.0:
         he_t_ms = 10 ** 8
     elif mass >= 2.0 and mass < 10.0:
         he_t_ms = 10**(-2.6094 * np.log10(mass) + 8.7855)
     elif mass >= 10.0 and mass < 100.0:
         he_t_ms = 10**(-0.69897 * np.log10(mass) + 6.875)
-    elif mass >= 100.0:
+    else: # mass >= 100.0
```

Comment:
may want to add a conditional posydonerror if the mass is not a number

Reply:
This is a general problem in many functions, that things are assumed and not checked for. I opened issue #462 .